src/Extensions/Testing.Helpers/NullableAssert.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Omex.Extensions.Testing.Helpers
+{
+	/// <summary>
+	/// Provider asserts checks with appropriate nullability attributes
+	/// </summary>
+	public static class NullableAssert
+	{
+		/// <summary>
+		/// Tests whether the specified object is non-null and throws an exception if it is null
+		/// </summary>
+		public static void IsNotNull([NotNull] object? value, string message = "", params object[] parameters) =>
+#pragma warning disable CS8777 // Assert.IsNotNull would throw if object is null
+			Assert.IsNotNull(value, message, parameters);
```

Comment:
Can't you use the ! symbol here?

Reply:
not it's a different warning it's for the whole method due to attribute application